src/app/wizard/step/provider-settings/provider/extended/azure/component.ts

```diff
@@ -95,17 +97,25 @@ export class AzureProviderExtendedComponent extends BaseFormValidator implements
       .pipe(takeUntil(this._unsubscribe))
       .subscribe(_ => {
         this._presets.enablePresets(
-          Object.values(this._clusterSpecService.cluster.spec.cloud.azure).every(value => !value)
+          Object.keys(this._clusterSpecService.cluster.spec.cloud.azure)
+            .filter(key => key !== 'assignAvailabilitySet')
+            .every(key => !this._clusterSpecService.cluster.spec.cloud.azure[key])
         );
       });
 
-    this._presets.presetChanges
-      .pipe(takeUntil(this._unsubscribe))
-      .subscribe(preset => Object.values(Controls).forEach(control => this._enable(!preset, control)));
+    this._presets.presetChanges.pipe(takeUntil(this._unsubscribe)).subscribe(preset =>
+      Object.values(Controls)
+        .filter(control => control !== Controls.AssignAvailabilitySet)
+        .forEach(control => this._enable(!preset, control))
+    );
 
     merge(this._clusterSpecService.providerChanges, this._clusterSpecService.datacenterChanges)
       .pipe(takeUntil(this._unsubscribe))
-      .subscribe(_ => this.form.reset());
+      .subscribe(_ => {
+        // TODO: This causes reset of all default values set in the init stage. It should be fixed.
```

Comment:
In case you haven't done yet: Can you please open an issue, so that we don't forget about it?

Reply:
Sure, I will do.